sumologic/resource_sumologic_muting_schedules_library_muting_schedule.go

```diff
@@ -0,0 +1,324 @@
+package sumologic
+
+import (
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceSumologicMutingSchedulesLibraryMutingSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicMutingSchedulesLibraryMutingScheduleCreate,
+		Read:   resourceSumologicMutingSchedulesLibraryMutingScheduleRead,
+		Update: resourceSumologicMutingSchedulesLibraryMutingScheduleUpdate,
+		Delete: resourceSumologicMutingSchedulesLibraryMutingScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: getMutingScheduleSchema(),
+	}
+}
+
+func getMutingScheduleBaseSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateFunc: validation.All(
+				validation.StringDoesNotContainAny("/"),
+				validation.StringLenBetween(1, 255),
+				validation.StringMatch(regexp.MustCompile(`(?s)^[^\ ].*[^\ ]$`),
+					"name must not contain leading or trailing spaces"),
+			),
+		},
+
+		"description": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(0, 4096),
+				validation.StringMatch(regexp.MustCompile(`(?s)^[^\ ].*[^\ ]$`),
+					"description must not contain leading or trailing spaces"),
+			),
+		},
+
+		"parent_id": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+
+		"monitor": {
+			Type:     schema.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: getMonitorScopeSchemma(),
+			},
+		},
+
+		"schedule": {
+			Type:     schema.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: getScheduleDefinitionSchemma(),
+			},
+		},
+
+		"version": {
+			Type:     schema.TypeInt,
+			Optional: true,
+			Computed: true,
+		},
+	}
+}
+
+func getMonitorScopeSchemma() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"ids": {
```

Comment:
lets add validation here to ensure ids/ all (true) is provided. many users complained that not having proper validation caused them issues.

Reply:
can I have the details of the validation? @rmamilla-sumo do you mean either "ids" or "all" field values need to be provided?
or something else?

Reply:
look @ AtLeastOneOf: triggerConditionsAtleastOneKey in monitors

Reply:
will update it in validation ticket.